src/utils/spawnAsync.ts

```diff
@@ -13,7 +13,7 @@ const DEFAULT_BUFFER_SIZE = 10 * 1024; // The default Node.js `exec` buffer size
 export type Progress = (content: string, process: cp.ChildProcess) => void;
 
 /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
-type ExecError = Error & { code: any, signal: any };
+export type ExecError = Error & { code: any, signal: any, stdErrHandled: boolean };

```

Comment:
The caller is passing the stdErr and we should leave the caller to make this call, than passing this information in the error.

Reply:
The caller here can choose to provide an strErrHandler or not (for example, DockerTaskProvider provides one while CliDockerClient doesn't). The thing is, the caller sometimes decides to write back the errorMessage which is thrown from here.

This flag simply lets the caller know that somewhere along the process lifetime, stdErrors were sent to the handler so no need to write them again.